### PR TITLE
BH-681: Upgrade to Symfony 5.3

### PR DESCRIPTION
**Pull request by jmleroux**
This PR will not be merged and will be split in smaller PRs

### CODE REVIEW

File: config/packages/test/messenger.yml [hunk 1]
```diff
@@ -3,57 +3,47 @@ framework:
         transports:
             # Producers
             business_event:
-                dsn: 'gps:'
+                dsn: doctrine://default
```
Comment (BitOne) [inline config/packages/test/messenger.yml:6]: Why not using GPS on test? Because community?

File: config/services/pim_parameters.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ parameters:
     pim_catalog_image_allowed_extensions: [gif, jfif, jif, jpeg, jpg, pdf, png, psd, tif, tiff]
     pim_catalog_file_allowed_extensions: [csv, doc, docx, mp3, pdf, ppt, pptx, rtf, svg, txt, wav]
 
-    doctrine_migrations_dir: '%kernel.root_dir%/../upgrades/schema'
+    doctrine_migrations_dir: '%kernel.project_dir%/upgrades/schema'
```
Comment (BitOne) [inline config/services/pim_parameters.yml:18]: project_dir already exists in 4.4, so it could be in independent PR before switching to 5.0

File: src/Akeneo/Channel/.php_cd.php [hunk 3]
```diff
@@ -26,7 +26,9 @@
     ])->in('Akeneo\Channel\Bundle'),
     $builder->only([
         'Symfony\Component',
+        'Symfony\Contracts\EventDispatcher\Event',
         'Doctrine\Common',
+        'Doctrine\Persistence',
```
Comment (BitOne) [inline src/Akeneo/Channel/.php_cd.php:31]: Harmeless if done now?

File: src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php [hunk 4]
```diff
@@ -56,12 +56,12 @@ public function remove($object, array $options = [])
 
         $options['unitary'] = true;
 
-        $this->eventDispatcher->dispatch(StorageEvents::PRE_REMOVE, new RemoveEvent($object, $objectId, $options));
+        $this->eventDispatcher->dispatch(new RemoveEvent($object, $objectId, $options), StorageEvents::PRE_REMOVE);
```
Comment (BitOne) [inline src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php:59]: Could be managed by:

https://github.com/rectorphp/rector-symfony/blob/main/docs/rector_rules_overview.md#makedispatchfirstargumenteventrector

File: src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php [hunk 5]
```diff
@@ -7,8 +7,8 @@
 use Akeneo\Tool\Component\StorageUtils\Event\RemoveEvent;
 use Akeneo\Tool\Component\StorageUtils\Remover\RemoverInterface;
 use Akeneo\Tool\Component\StorageUtils\StorageEvents;
-use Doctrine\Common\Persistence\ObjectManager;
 use Doctrine\Common\Util\ClassUtils;
+use Doctrine\Persistence\ObjectManager;
```
Comment (BitOne) [inline src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php:11]: A simple rector rules could do that.

File: src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php [hunk 6]
```diff
@@ -83,16 +83,20 @@ private function validateObject($object)
 
         $channelCount = $this->channelRepository->countAll();
         if (1 === $channelCount) {
-            throw new \LogicException($this->translator->trans(
-                'pim_enrich.channel.flash.delete.error',
-                ['%channelCode%' => $object->getCode() ]
-            ));
+            throw new \LogicException(
```
Comment (BitOne) [inline src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php:86]: Maybe reformatting should not in this PR? Not related to SF5.

File: src/Akeneo/Channel/Bundle/Resources/config/view_elements/attribute.yml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 parameters:
-    pim_enrich.view_element.attribute.tab.history.template:   'PimUIBundle:Form:Tab/history.html.twig'
+    pim_enrich.view_element.attribute.tab.history.template:   '@@PimUI/Form/Tab/history.html.twig'
```
Comment (BitOne) [inline src/Akeneo/Channel/Bundle/Resources/config/view_elements/attribute.yml:2]: Why double @?

File: src/Akeneo/Channel/Bundle/Twig/LocaleExtension.php [hunk 8]
```diff
@@ -94,7 +94,7 @@ public function currencySymbol($code, $translateIn = null)
         $translateIn = $translateIn ?: $this->getCurrentLocaleCode();
         $language = \Locale::getPrimaryLanguage($translateIn);
 
-        return Intl\Intl::getCurrencyBundle()->getCurrencySymbol($code, $language);
+        return Intl\Currencies::getSymbol($code, $language);
```
Comment (BitOne) [inline src/Akeneo/Channel/Bundle/Twig/LocaleExtension.php:97]: Rector rule.

File: src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 namespace Akeneo\Channel\Component\Event;
 
-use Symfony\Component\EventDispatcher\Event;
+use Symfony\Contracts\EventDispatcher\Event;
```
Comment (BitOne) [inline src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php:7]: Maybe in SF 4.4?

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookUserAuthenticator.php [hunk 10]
```diff
@@ -15,11 +16,10 @@
  */
 class WebhookUserAuthenticator
 {
-    private ObjectRepository $userRepository;
-
+    private UserRepositoryInterface $userRepository;
     private TokenStorageInterface $tokenStorage;
 
-    public function __construct(ObjectRepository $userRepository, TokenStorageInterface $tokenStorage)
+    public function __construct(UserRepositoryInterface $userRepository, TokenStorageInterface $tokenStorage)
```
Comment (BitOne) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookUserAuthenticator.php:22]: Nice change, but not SF 5 specific?

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ApiErrorEventSubscriberSpec.php [hunk 11]
```diff
@@ -27,12 +26,14 @@ public function let(CollectApiError $collectApiError): void
...
-        $this->getSubscribedEvents()->shouldReturn([
-            ProductDomainErrorEvent::class => 'collectProductDomainError',
-            ProductValidationErrorEvent::class => 'collectProductValidationError',
-            TechnicalErrorEvent::class => 'collectTechnicalError',
-            KernelEvents::TERMINATE => 'flushApiErrors',
-        ]);
+        $this->getSubscribedEvents()->shouldReturn(
+            [
+                ProductDomainErrorEvent::class => 'collectProductDomainError',
+                ProductValidationErrorEvent::class => 'collectProductValidationError',
```
Comment (BitOne) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ApiErrorEventSubscriberSpec.php:32]: Fomatting?

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php [hunk 12]
```diff
@@ -14,7 +14,7 @@
 use Akeneo\UserManagement\Bundle\Context\UserContext;
 use Doctrine\Common\Collections\ArrayCollection;
 use Oro\Bundle\SecurityBundle\SecurityFacade;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
```
Comment (BitOne) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php:17]: Rector rule.

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php [hunk 13]
```diff
@@ -142,7 +142,7 @@ public function listTreeAction(Request $request): Response
         }
 
         return $this->render(
-            'AkeneoPimEnrichmentBundle:CategoryTree:listTree.json.twig',
+            '@AkeneoPimEnrichment/CategoryTree/listTree.json.twig',
```
Comment (BitOne) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php:145]: Working in SF 4.4?

File: .php_cs.php [hunk 14]
```diff
@@ -1,14 +1,16 @@
...
-        ],
-    ))
+return (new PhpCsFixer\Config())
+    ->setRules(
+        [
+            '@PSR2' => true,
+            'linebreak_after_opening_tag' => true,
+            'ordered_imports' => true,
+            'method_argument_space' => [
+                'on_multiline' => 'ignore',
```
Comment (BitOne) [inline .php_cs.php:10]: Dedicated PR?

File: composer.json [hunk 15]
```diff
@@ -47,117 +48,115 @@
...
+        "doctrine/orm": "^2.9.3",
+        "doctrine/persistence": "^2.2.1",
+        "doctrine/reflection": "^1.2.2",
+        "dompdf/dompdf": "1.0.2",
         "elasticsearch/elasticsearch": "7.11.0",
         "friendsofsymfony/jsrouting-bundle": "2.7.0",
-        "friendsofsymfony/oauth-server-bundle": "1.6.2",
-        "friendsofsymfony/rest-bundle": "2.8.6",
-        "gedmo/doctrine-extensions":"v3.0.3",
+        "klapaudius/oauth-server-bundle": "3.0.0",
```
Comment (jmleroux) [inline composer.json:77]: **WARNING*: check for a better/safer way to handle OAuth server to not use a forked library

File: composer.json [hunk 16]
```diff
@@ -47,117 +48,115 @@
...
-        "behat/mink-browserkit-driver": "1.3.3",
-        "behat/mink-extension": "2.3.1",
+        "behat/gherkin": "4.6.2",
         "behat/mink-selenium2-driver": "^1.3.1",
+        "behat/transliterator": "1.2.0",
+        "dvdoug/behat-code-coverage": "^5.2",
+        "friends-of-behat/mink": "1.7.1",
+        "friends-of-behat/mink-browserkit-driver": "1.3.3",
+        "friends-of-behat/mink-extension": "^2.5.0",
+        "friends-of-behat/symfony-extension": "^2.2",
```
Comment (jmleroux) [inline composer.json:142]: **WARNING**: check this library for Behat

File: config/packages/doctrine.yml [hunk 17]
```diff
@@ -31,7 +31,7 @@ doctrine:
                 type: annotation
                 alias: Gedmo
                 prefix: Gedmo\Tree\Entity
-                dir: '%kernel.root_dir%/../vendor/gedmo/doctrine-extensions/src/Tree/Entity'
+                dir: '%kernel.project_dir%/vendor/gedmo/doctrine-extensions/src/Tree/Entity'
```
Comment (jmleroux) [inline config/packages/doctrine.yml:34]: **PREMIGRATION**: can be done before migration

File: config/packages/doctrine.yml [hunk 18]
```diff
@@ -56,6 +56,10 @@ framework:
...
 doctrine_migrations:
-    dir_name: '%doctrine_migrations_dir%'
-    namespace: Pim\Upgrade\Schema
-    name: PIM Migrations
+    migrations_paths:
+        'Pim\Upgrade\Schema': '%kernel.project_dir%/upgrades/schema'
+    storage:
+        table_storage:
+            table_name: 'migration_versions'
+            version_column_length: 190
```
[the remark this answers is not in the document]
Comment (jmleroux) [reply config/packages/doctrine.yml:64]: We may not go above 190

File: config/packages/fos_rest.yml [hunk 19]
```diff
@@ -17,9 +16,8 @@ fos_rest:
     body_listener:
         decoders:
             json: fos_rest.decoder.json
-    routing_loader:
-        default_format: json
     exception:
+        serializer_error_renderer: true
```
Comment (jmleroux) [inline config/packages/fos_rest.yml:20]: **PREMIGRATION** update before migration

File: config/packages/oro_filter.yml [hunk 20]
```diff
@@ -1,3 +1,3 @@
 oro_filter:
     twig:
-        layout: PimFilterBundle:Filter:pim-layout.js.twig
+        layout: '@PimFilter/Filter/pim-layout.js.twig'
```
Comment (jmleroux) [inline config/packages/oro_filter.yml:3]: **PREMIGRATION** `:` notation is deprecated

File: config/packages/security.yml [hunk 21]
```diff
@@ -9,7 +9,12 @@ security:
             id: pim_user.provider.user_api
 
     encoders:
-        Akeneo\UserManagement\Component\Model\User: sha512
+        legacy_sha512:
```
Comment (jmleroux) [inline config/packages/security.yml:12]: **WARNING** new users created after migration will need password reset in case of rollback

File: config/packages/security.yml [hunk 22]
```diff
@@ -26,13 +31,11 @@ security:
             pattern:                        ^/user/(login|reset-request|send-email|check-email)$
             provider:                       chain_provider
             anonymous:                      true
-            logout_on_user_change:          true
```
Comment (jmleroux) [inline config/packages/security.yml:29]: **PREMIGRATION** deprecated in 4.1

File: config/routes/dev/routes.yml [hunk 23]
```diff
@@ -7,5 +7,5 @@ _profiler:
     prefix:   /_profiler
 
 _errors:
-    resource: "@TwigBundle/Resources/config/routing/errors.xml"
```
Comment (jmleroux) [inline config/routes/dev/routes.yml:10]: **PREMIGRATION** to check

File: src/Akeneo/Channel/.php_cd.php [hunk 24]
```diff
@@ -26,7 +26,9 @@
     ])->in('Akeneo\Channel\Bundle'),
     $builder->only([
         'Symfony\Component',
+        'Symfony\Contracts\EventDispatcher\Event',
```
Comment (jmleroux) [inline src/Akeneo/Channel/.php_cd.php:29]: **PREMIGRATION** to check

File: src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php (hunk 9 above)
Comment (jmleroux) [reply src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php:7]: **PREMIGRATION** to check

File: src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php (hunk 6 above)
Comment (jmleroux) [reply src/Akeneo/Channel/Bundle/Doctrine/Remover/ChannelRemover.php:86]: To check if compatible with CS fixer multiline in this version

File: src/Akeneo/Channel/.php_cd.php (hunk 24 above)
Comment (jmleroux) [reply src/Akeneo/Channel/.php_cd.php:29]: Checked by Benoit, doable in 4.4

File: src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php (hunk 9 above)
Comment (BitOne) [reply src/Akeneo/Channel/Component/Event/ChannelCategoryHasBeenUpdated.php:7]: Yes: https://github.com/symfony/contracts/blob/1.1/EventDispatcher/Event.php (we use sf contracts ^1.1.8)

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CheckAuthenticationSecurityEndToEnd.php [hunk 25]
```diff
@@ -65,6 +65,7 @@ public function test_the_secret_regeneration_disables_the_secret()
 
         $this->regenerateClientSecret('magento');
 
+        static::ensureKernelShutdown();
```
Comment (jmleroux) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CheckAuthenticationSecurityEndToEnd.php:68]: **PREMIGRATION** to check

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ApiErrorEventSubscriberSpec.php [hunk 26]
```diff
@@ -75,10 +76,10 @@ public function it_collects_a_technical_error($collectApiError): void
         $this->collectTechnicalError($event);
     }
 
-    public function it_flushes_collected_errors($collectApiError, TerminateEvent $terminateEvent): void
```
Comment (jmleroux) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ApiErrorEventSubscriberSpec.php:78]: **PREMIGRATION**

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/MessageHandler/BusinessEventHandlerSpec.php [hunk 27]
```diff
@@ -24,6 +25,7 @@ public function let(
         SendBusinessEventToWebhooksHandler $commandHandler,
         EventDispatcherInterface $eventDispatcher
     ): void {
+        $eventDispatcher->dispatch(Argument::type('object'))->willReturn(Argument::type('object'));
```
Comment (jmleroux) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/MessageHandler/BusinessEventHandlerSpec.php:28]: **PREMIGRATION** (with PHPSpec upgrade?)

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/AbstractListCategoryController.php [hunk 28]
```diff
@@ -6,7 +6,7 @@
 use Akeneo\Tool\Component\Classification\Repository\CategoryRepositoryInterface;
 use Doctrine\Common\Collections\Collection;
 use Oro\Bundle\SecurityBundle\SecurityFacade;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/AbstractListCategoryController.php:9]: **PREMIGRATION**

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/FileController.php [hunk 29]
```diff
@@ -9,12 +9,12 @@
 use Akeneo\Tool\Component\FileStorage\Repository\FileInfoRepositoryInterface;
 use Akeneo\Tool\Component\FileStorage\StreamedFileResponse;
 use Liip\ImagineBundle\Controller\ImagineController;
-use Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesser;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Mime\MimeTypes;
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/FileController.php:17]: **PREMIGRATION**

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/FileController.php [hunk 30]
```diff
@@ -26,32 +26,13 @@ class FileController
...
     const SVG_MIME_TYPES = ['image/svg', 'image/svg+xml'];
 
-    /** @var ImagineController */
-    protected $imagineController;
+    protected ImagineController $imagineController;
+    protected FilesystemProvider $filesystemProvider;
+    protected FileInfoRepositoryInterface $fileInfoRepository;
+    protected FileTypeGuesserInterface $fileTypeGuesser;
+    protected DefaultImageProviderInterface $defaultImageProvider;
+    protected array $filesystemAliases;
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/FileController.php:34]: Could be rectorized

Comment (jmleroux) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/FileController.php:34]: Rector PHP 7.4

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ReferenceDataRepositoryResolver.php [hunk 31]
```diff
@@ -15,28 +16,18 @@
...
-    protected $doctrineRegistry;
-
-    /**
-     * @param ConfigurationRegistryInterface $configurationRegistry
-     * @param RegistryInterface              $doctrineRegistry
-     */
     public function __construct(
         ConfigurationRegistryInterface $configurationRegistry,
-        RegistryInterface $doctrineRegistry
+        Registry $doctrineRegistry
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ReferenceDataRepositoryResolver.php:24]: **PREMIGRATION** ?

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/doctrine/Category/Category.orm.yml [hunk 32]
```diff
@@ -12,9 +12,6 @@ Akeneo\Pim\Enrichment\Component\Category\Model\Category:
         left_idx:
             columns:
                 - lft
-    gedmo:
-        tree:
-            type: nested
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Resources/config/doctrine/Category/Category.orm.yml:17]: **PREMIGRATION** Doctrine mappiung uin YAML is deprecated. Use XML or Annotations

File: src/Akeneo/Pim/Enrichment/Component/Category/Model/Category.php [hunk 33]
```diff
@@ -36,11 +40,26 @@ class Category extends BaseCategory implements CategoryInterface
     /** @var ArrayCollection of Channel */
     protected $channels;
 
-    /** @var \DateTime */
+    /**
+     * @var \DateTime
+     * @Gedmo\Timestampable(on="create")
+     */
     protected $created;
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Component/Category/Model/Category.php:47]: Type hint

File: src/Akeneo/Pim/Enrichment/Component/Product/Repository/ReferenceDataRepositoryResolverInterface.php [hunk 34]
```diff
@@ -11,10 +13,5 @@
  */
 interface ReferenceDataRepositoryResolverInterface
 {
-    /**
-     * @param string $referenceDataType
-     *
-     * @return \Doctrine\Common\Persistence\ObjectRepository
-     */
-    public function resolve($referenceDataType);
+    public function resolve(string $referenceDataType): ObjectRepository;
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Component/Product/Repository/ReferenceDataRepositoryResolverInterface.php:16]: Tom ake in another PR

File: src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/DateValidator.php [hunk 35]
```diff
@@ -21,6 +21,10 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, Date::class);
         }
 
+        if ($value instanceof \DateTimeInterface) {
+            return;
+        }
+
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/DateValidator.php:27]: **PREMIGRATION** to check

File: src/Akeneo/Pim/Structure/Bundle/DependencyInjection/Configuration.php [hunk 36]
```diff
@@ -19,8 +19,8 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('akeneo_pim_structure');
+        $treeBuilder = new TreeBuilder('akeneo_pim_structure');
+        $rootNode = $treeBuilder->getRootNode();
```
Comment (jmleroux) [inline src/Akeneo/Pim/Structure/Bundle/DependencyInjection/Configuration.php:23]: **PREMIGRATION** check

File: src/Akeneo/Platform/Bundle/CatalogVolumeMonitoringBundle/Command/AggregateVolumesCommand.php [hunk 37]
```diff
@@ -35,19 +34,20 @@ public function __construct(VolumeAggregation $volumeAggregation)
...
     {
-        $this
-            ->setDescription('Aggregate the result of all the volume queries that should not be executed live');
+        $this->setDescription('Aggregate the result of all the volume queries that should not be executed live');
     }
 
     /**
      * {@inheritdoc}
      */
-    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/CatalogVolumeMonitoringBundle/Command/AggregateVolumesCommand.php:45]: **PREMIGRATION**

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/back/Infrastructure/Delivery/InternalApi/Announcement/AddViewedAnnouncementsAction.php [hunk 38]
```diff
@@ -43,7 +40,7 @@ public function __invoke(Request $request)
         }
 
         $command = new AddViewedAnnouncementsByUserCommand(
-            $request->request->get('viewed_announcement_ids'),
+            (array) $request->request->get('viewed_announcement_ids'),
             $user->getId()
         );
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/back/Infrastructure/Delivery/InternalApi/Announcement/AddViewedAnnouncementsAction.php:45]: Code smell

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Normalizer/StepExecutionNormalizer.php [hunk 39]
```diff
@@ -112,7 +115,7 @@ protected function normalizeSummary(array $summary)
         $result = [];
         foreach ($summary as $key => $value) {
             $key = sprintf('job_execution.summary.%s', $key);
-            $result[$this->translator->trans($key)] = $this->translator->trans($value);
+            $result[$this->translator->trans($key)] = $this->translator->trans((string)$value);
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Normalizer/StepExecutionNormalizer.php:118]: Code smell

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/config/controllers.yml [hunk 40]
```diff
@@ -72,6 +72,8 @@ services:
             - '@akeneo_batch.job.job_registry'
             - '@logger'
             - '@oneup_flysystem.archivist_filesystem'
+        calls:
+            - [ setContainer, [ '@service_container' ] ]
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/config/controllers.yml:76]: Try to remove this call.

If not possible, study:
- ContainerAware trait
- ContainerAwareInterface

File: src/Akeneo/Platform/Bundle/InstallerBundle/Command/DatabaseCommand.php [hunk 41]
```diff
@@ -249,33 +227,9 @@ protected function createNotMappedTables(OutputInterface $output)
                 `values` JSON NOT NULL
             ) COLLATE utf8mb4_unicode_ci, ENGINE = InnoDB;";
         $this->connection->exec($configTableSql);
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/InstallerBundle/Command/DatabaseCommand.php:229]: TR

File: src/Akeneo/Tool/Bundle/StorageUtilsBundle/Doctrine/MappingsOverrideConfiguratorInterface.php [hunk 42]
```diff
@@ -24,12 +24,6 @@ interface MappingsOverrideConfiguratorInterface
...
      *  ]
-     *
-     * @param ClassMetadata $metadata
-     * @param mixed         $configuration
-     * @param array         $mappingOverrides
-     *
-     * @return ClassMetadata
      */
-    public function configure(ClassMetadata $metadata, array $mappingOverrides, $configuration);
+    public function configure(ClassMetadata $metadata, array $mappingOverrides, $configuration): ClassMetadata;
```
Comment (jmleroux) [inline src/Akeneo/Tool/Bundle/StorageUtilsBundle/Doctrine/MappingsOverrideConfiguratorInterface.php:28]: **PREMIGRATION**
